userapi/src/main/java/fk/prof/userapi/verticles/HttpVerticle.java

```diff
@@ -127,16 +129,17 @@ private void getProfiles(RoutingContext routingContext) {
         try {
             startTime = ZonedDateTime.parse(routingContext.request().getParam("start"), DateTimeFormatter.ISO_ZONED_DATE_TIME);
             duration = Integer.parseInt(routingContext.request().getParam("duration"));
-        }
-        catch (Exception e) {
+            if (duration > MAX_TIME_WINDOW) { // Seconds in seven days
+                throw new IllegalArgumentException("Max window size supported = " + MAX_TIME_WINDOW + " seconds, requested window size = " + duration + " seconds");
```

Comment:
you dont really need to follow throw catch here. 
Could have just created a exception and passed it as param to setresponse.
Keep try catch for unexpected cases.